### PR TITLE
Adding some more handling for recognizing client disconnects

### PR DESCRIPTION
**Pull request by vgkholla**

### CODE REVIEW

File: ambry-rest/src/main/java/com.github.ambry.rest/NettyResponseChannel.java [hunk 1]
```diff
@@ -388,7 +386,7 @@ private FullHttpResponse getErrorResponse(Throwable cause) {
             Utils.getRootCause(cause).getMessage().replaceAll("[\n\t\r]", " ").getBytes(StandardCharsets.US_ASCII),
             StandardCharsets.US_ASCII);
       }
-    } else if (isPossibleClientTerminate(cause)) {
+    } else if (Utils.isPossibleClientTerminate(cause)) {
```
Comment (pnarayanan) [inline ambry-rest/src/main/java/com.github.ambry.rest/NettyResponseChannel.java]: minor: "client termination"? ("terminate" sounds a bit odd as a noun).

File: ambry-utils/src/test/java/com.github.ambry.utils/UtilsTest.java [hunk 2]
```diff
@@ -420,6 +420,27 @@ public void addSecondsToEpochTimeTest() {
     }
   }
 
+  /**
+   * Tests for {@link Utils#isPossibleClientTerminate(Throwable)} and
+   * {@link Utils#convertToClientTerminateException(Throwable)}.
+   */
+  @Test
+  public void clientTerminateWrapAndRecognizeTest() {
+    Exception exception = new IOException("Connection reset by peer");
```
Comment (pnarayanan) [inline ambry-utils/src/test/java/com.github.ambry.utils/UtilsTest.java]: Use the constant?

File: ambry-utils/src/test/java/com.github.ambry.utils/UtilsTest.java [hunk 3]
```diff
@@ -420,6 +420,27 @@ public void addSecondsToEpochTimeTest() {
...
+    assertTrue("Should be declared as a client terminate", Utils.isPossibleClientTerminate(exception));
+
+    exception = new IOException("Connection not reset by peer");
+    assertFalse("Should not be declared as a client terminate", Utils.isPossibleClientTerminate(exception));
+    exception = Utils.convertToClientTerminateException(exception);
+    assertTrue("Should be declared as a client terminate", Utils.isPossibleClientTerminate(exception));
+
+    exception = new Exception("Connection reset by peer");
```
Comment (pnarayanan) [inline ambry-utils/src/test/java/com.github.ambry.utils/UtilsTest.java]: Use the constant?

File: ambry-utils/src/test/java/com.github.ambry.utils/UtilsTest.java (hunk 2 above)
Comment (vgkholla) [reply ambry-utils/src/test/java/com.github.ambry.utils/UtilsTest.java]: Discussed offline

File: ambry-utils/src/test/java/com.github.ambry.utils/UtilsTest.java (hunk 3 above)
Comment (vgkholla) [reply ambry-utils/src/test/java/com.github.ambry.utils/UtilsTest.java]: Discussed offline

**Pull request merged**